src/rgw/rgw_op.h

```diff
@@ -300,6 +302,7 @@ class RGWOp : public DoutPrefixProvider {
   }
   virtual const char* name() const = 0;
   virtual RGWOpType get_type() { return RGW_OP_UNKNOWN; }
+  virtual const std::string canonical_name() const { return name(); }
```

Comment:
for the base class `RGWOp` we can have default implementation:
```
return fmt::format("REST.{}.{}", get_http_method_str(), name());
```

but, ideally, we put the correct "resource" in each derived op (which is not the same as `name()`)